### PR TITLE
{AKS} `az aks update`: refactor pod_cidr and network_plugin_mode getters to check decorator mode

### PR DESCRIPTION
**Pull request by tyler-lloyd**
add checks for decorator mode to better match the
logic in the acs (GA CLI) and avoid reading any values from mc unless doing CREATE

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az aks update`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -3242,7 +3240,7 @@ def update_upgrade_settings(self, mc: ManagedCluster) -> ManagedCluster:
             if override_until is not None:
                 try:
                     mc.upgrade_settings.override_settings.until = parse(override_until)
-                except Exception as ex:  # pylint: disable=broad-except
+                except Exception as ex:  # pylint: disable=broad-except # noqa: F841
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3243]:
```suggestion
                except Exception:  # pylint: disable=broad-except
```

Maybe something like this

**Review by FumingZhang — COMMENTED** [review]
LGTM

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
Comment (tyler-lloyd) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3243]: also fine with that 👍

**Comment by tyler-lloyd** [comment]
Added note in history under pending section

**Comment by tyler-lloyd** [comment]
@zhoxing-ms PTAL

**Review by zhoxing-ms — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]
LGTM

**Pull request merged**